src/blocks/class-blocks.php

```diff
@@ -37,10 +37,20 @@ public function register() {
    */
   public function get_all_allowed_forms_blocks( $allowed_block_types, $post ) {
     if ( $post->post_type === Forms::POST_TYPE_SLUG ) {
+
+      // Remove forms select on form builder post type.
+      $allInternalBlock = $this->get_all_blocks_list();
+      if (($key = array_search('eightshift-forms/forms', $allInternalBlock, true)) !== false) {
```

Comment:
Assignment inside a conditional should be avoided at all costs. It's hard to read and prone to errors.

The assignment to `$key` should be extracted out of the conditional.

Also, I'm not 100% sure what the `$allInternalBlock` returns, but if it's a simple array,
```php
isset($allInternalBlock['eightshift-forms/forms'])
``` 
should do the trick. Plus it's way faster than an `array_search()` call.